browser/views.py

```diff
@@ -713,6 +713,67 @@ def delete_investigator_record(request, person_id, research_id):
     return HttpResponseRedirect(reverse('person', args=(person_id,)))
 
 
+@staff_member_required
+def edit_affiliation(request, person_id, affiliation_id):
+    person = get_object_or_404(Person, pk=person_id)
+    affiliation = get_object_or_404(Affiliation, pk=affiliation_id)
+    template = "browser/affiliation_edit.html"
+
+    form = EditAffiliationForm(initial={'institution': affiliation.institution,
+                                        'position': affiliation.position,
+                                        'institution_id': affiliation.institution.id if affiliation.institution else None})
+    context = {
+        'form': form,
+        'person': person
+    }
+
+    if request.method == 'POST':
+        form = EditAffiliationForm(request.POST, instance=person)
+        success, institution = _handle_institution(form, request)
+        if success:
+            Affiliation.objects.filter(id=affiliation_id).update(position=form.cleaned_data.get('position'),
+                                                     institution=institution)
```

Comment:
is there a reason you don't use save() here? If there are any actions associated with saving an object (e.g. update last modified timestamp), these wont' be run if you do it this way.